pyscript.core/src/stdlib/pyweb/pydom.py

```diff
@@ -0,0 +1,420 @@
+import inspect
+import sys
+from functools import cached_property
+from typing import Any
+
+import js
+
+# from js import document as js_document
+from pyodide.ffi import JsProxy
+from pyodide.ffi.wrappers import add_event_listener
+from pyscript import display
+
+# from pyscript import when as _when
+
+alert = js.alert
+
+
+class BaseElement:
+    def __init__(self, js_element):
+        self._element = js_element
+        self._parent = None
+        self.style = StyleProxy(self)
+
+    def __eq__(self, obj):
+        """Check if the element is the same as the other element by comparing
+        the underlying JS element"""
+        return isinstance(obj, BaseElement) and obj._element == self._element
+
+    @property
+    def parent(self):
+        if self._parent:
+            return self._parent
+
+        if self._element.parentElement:
```

Comment:
just to clarify: `parentElement` breaks if parent is a fragment or a document ... if that's intended, all good!

Reply:
Not sure I understand... how does it break? If the underlying `_element` here is a fragment or a document, the value of `self._element.parentElement` (hence the value of the `parent` property) is None and returns `None`

Reply:
the DOM has both `parentNode` and `parentElement` but here we use the latter but not the former. If that's meant, it's all good, but as no explicit comment was in here I just wondered if that was the case.